recipes/system-probe.rb

```diff
@@ -0,0 +1,58 @@
+#
+# Cookbook Name:: datadog
+# Recipe:: system-probe
+#
+# Copyright 2011-2019, Datadog
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Set the correct agent startup action
+sysprobe_agent_start = node['datadog']['system_probe']['enabled'] ? :start : :stop
+
+#
+# Configures system-probe agent
+system_probe_config_file = ::File.join(node['datadog']['agent6_config_dir'], 'system-probe.yaml')
+template system_probe_config_file do
+  source 'system_probe.yaml.erb'
+  variables(
+    enabled: node['datadog']['system_probe']['enabled'],
+    sysprobe_socket: node['datadog']['system_probe']['sysprobe_socket'],
+    debug_port: node['datadog']['system_probe']['debug_port'],
+    bpf_debug: node['datadog']['system_probe']['bpf_debug'],
+    enable_conntrack: node['datadog']['system_probe']['enable_conntrack']
+  )
+  owner 'dd-agent'
+  group 'root'
+  mode '640'
+  notifies :restart, 'service[datadog-agent-sysprobe]', :delayed unless node['datadog']['system_probe']['enabled'] == false
+  # since process-agent collects network info through system-probe, enabling system-probe should also restart process-agent
+  notifies :restart, 'service[datadog-agent]', :delayed unless node['datadog']['system_probe']['enabled'] == false
+end
+
+# Common configuration
+service_provider = nil
+if node['datadog']['agent6'] &&
+   (((node['platform'] == 'amazon' || node['platform_family'] == 'amazon') && node['platform_version'].to_i != 2) ||
+    (node['platform'] == 'ubuntu' && node['platform_version'].to_f < 15.04) || # chef <11.14 doesn't use the correct service provider
+   (node['platform'] != 'amazon' && node['platform_family'] == 'rhel' && node['platform_version'].to_i < 7))
+  # use Upstart provider explicitly for Agent 6 on Amazon Linux < 2.0 and RHEL < 7
+  service_provider = Chef::Provider::Service::Upstart
```

Comment:
Shouldn't there be a case where we use systemd?

Reply:
I think if it's nil it's going to use the default, https://github.com/DataDog/chef-datadog/pull/626/files#diff-a9b4a2e1fcc3717856f3e54e31a25c38R55, and according to chef doc that's the preferable way: 
> Generally, it’s best to let the chef-client choose the provider and this is (by far) the most common approach.